autodoc.pl

```diff
@@ -2353,7 +2378,45 @@ ($)
     next if $file =~ m! ^ ( cpan | dist | ext ) / !x
          && ! defined $extra_input_pods{$file};
 
-    if ($file =~ /\.h/) {
+    # Process these two special files immediately.  Otherwise we add the file to
+    # the appropriate list.
+    if ($file eq "proto.h") {
+
+        # proto.h won't have any apidoc lines in it.  Instead look for real
+        # prototypes.  Then we can check later that a prototype actually
+        # exists when we add a line to the pod that claims there is.
+        open my $fh, '<', $file or die "Cannot open $file for docs: $!\n";
+        my $prev = "";
+        while (defined (my $input = <$fh>)) {
+
+            # Look for a prototype.  As an extra little nicety, make sure that
+            # the line previous to the prototype is one of the one that
+            # declares the return type of the function.  This is to try to
+            # eliminate false positives.
+            $protos{$1} = $2 if $prev =~ /  ^ \s*
+                PERL_ (?: CALLCONV
+                                                    | STATIC (?: _FORCE)? _INLINE
+                                                  )
+                                         /x
+                           && $input =~ s/ ^ \s* ( [Pp]erl_\w* ) ( .* ) \n/$1/x;
```

Comment:
the indentation here is all over the place

Reply:
I don't know how that happened.  But now corrected, and the order of the two matches reversed, so that $1 and $2 are closer to where they are used